others/kadanes3.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * @file
+ * @brief Efficient implementation for maximum contiguous subarray sum by
+ * [Kadane's
+ * algorithm](https://www.geeksforgeeks.org/largest-sum-contiguous-subarray/).
+ * @details
+ * Our task is to take length of array and then the whole array as input from
+ * the user and then calculate the maximum contiguos subarray sum for the
+ * input array, using the kadane's algorithm.
+ *
+ * There can be a case that all the elements in the input array are negative.
+ * In that case, the least value among all elements is the maximum sum with
+ * subarray length = 1.
+ * @author [Abhijeet Tiwari](https://github.com/thisabhijeet)
+ */
+
+#include <array>     /// for std::array
+#include <cassert>   /// for assert
+#include <climits>   /// for INT_MIN value
+#include <iostream>  /// for IO operations
+
+/**
+ * @brief Utility function to check the current maximum number
+ * \param arr input array
+ * \param length length of the input array
+ * \returns maximum contiguous subarray sum
+ */
+template <std::size_t SIZE>
+int max_subarray_sum(std::array<int, SIZE> arr, int length) {
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂